htmd/smallmol/smallmol.py

```
@@ -1,105 +1,654 @@
+                raise ValueError("Unkown '{}' provided. Not a valid mol2,pdb,smile, rdkitMol obj. "
+                                 "Try by setting the force_reading option as True.".format(mol))
+            else:
+                raise ValueError("Unkown '{}' provided. Not a valid mol2,pdb,smile, rdkitMol obj.".format(mol))
+
+        if removeHs:
+            self._mol = Chem.RemoveHs(self._mol)
+
+        if fixHs:
+            self._mol = Chem.AddHs(self._mol, addCoords=True)
+
+        # fill the molecule and atom properties
+        self._readMol(smallMol)
+
+    @property
+    def listProps(self):
+        """
+        The list of properties of the molecule
+
+        Returns
+        -------
+        fields: list
+            The list of properties
+        """
+        _fields = list(self.mol_fields)
+        return _fields
+
+    @property
+    def listPropsAtoms(self):
+        """
+        The list of properties of the atoms
+
+        Returns
+        -------
+        fields: list
+            The list of properties
+        """
+        _fields = list(self.atom_fields)
+        return _fields
 
-    def __init__(self, mol, ignore_errors=False, addHs=True):
+    @property
+    def numAtoms(self):
         """
-        Initializes small molecule object
+        Returns the number of atoms in the molecule
+
+        Returns
+        -------
+        natoms: int
+            The number of atoms in the molecule
+
+        """
+        # getting the number of atoms from the shape of element array
+        natoms = self.element.shape[0]
+        return natoms
+
+
+    @property
+    def numConformers(self):
+        """
+        Returns the number of conformers in the rdkit molecule object
+
+        Returns
+        -------
+        nconfs: int
+            The number of conformers in the molecule
+        """
+        # getting the number of conformers from the shape of coords array
+        _nConformers = self.coords.shape[-1]
```

detail: I'd prefer it returning `self.coords.shape[2]`. That way if for some reason you messed up and `self.coords` is two dimensional it will crash instead of returning `3` for `xyz`

---

I see. thanks for the suggestion. Done it